src/reduxFiles/reducers/watch-later/watchLaterSlice.js

```diff
@@ -0,0 +1,100 @@
+import { createSlice, createAsyncThunk } from "@reduxjs/toolkit";
+import {
+  addVideoToWatchLaterService,
+  receiveAllWatchLaterVideosService,
+  removeVideoFromWatchLaterService,
+} from "services";
+const initialState = {
+  status: null,
+  watchLaterVideosList: [],
+};
+
+const receiveAllWatchLaterVideosHandler = async (tokenProvided) => {
+  try {
+    const {
+      data: { watchlater: watchLaterVideosResponseReceived },
+    } = await receiveAllWatchLaterVideosService(tokenProvided);
+    return watchLaterVideosResponseReceived;
+  } catch (receiveAllWatchLaterVideosError) {
+    console.error(
+      "AN ERROR OCCURED WHILE RECEIVING ALL WATCH LATER VIDEOS THROUGH REDUX"
+    );
+    return Promise.error(receiveAllWatchLaterVideosError);
+  }
+};
+
+const addNewVideoToWatchLaterHandler = async ({
+  videoDetailsGiven,
+  tokenProvided,
+}) => {
+  try {
+    const {
+      data: { watchlater: watchLaterVideosResponseReceived },
+    } = await addVideoToWatchLaterService(videoDetailsGiven, tokenProvided);
+    return watchLaterVideosResponseReceived;
+  } catch (addNewVideoToWatchLaterError) {
+    console.error(
+      "AN ERROR OCCURED WHILE ADDING A VIDEO TO WATCH LATER SECTION THROUGH REDUX"
+    );
+    return Promise.error(addNewVideoToWatchLaterError);
```

Comment:
suggestion: if you use rejectWithThunk("error") here, it will add this error in the payload and in the `rejected` addcase you can access that using `action.payload`

Reply:
I have tried using these changes in the component where I am dispatching the action